conductor-client/src/main/kotlin/com/openlattice/authorization/HazelcastPrincipalsMapManager.kt

```diff
@@ -0,0 +1,61 @@
+package com.openlattice.authorization
+
+import com.google.common.base.Preconditions
+import com.hazelcast.core.HazelcastInstance
+import com.hazelcast.query.Predicate
+import com.hazelcast.query.Predicates
+import com.openlattice.authorization.mapstores.PrincipalMapstore
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.organization.roles.Role
+
+/**
+ * Principals manager manages the principals map
+ *
+ * @author Drew Bailey (drew@openlattice.com)
+ */
+class HazelcastPrincipalsMapManager(
+        hazelcastInstance: HazelcastInstance,
+        private val reservations: HazelcastAclKeyReservationService
+): PrincipalsMapManager {
+
+    private val principals = HazelcastMap.PRINCIPALS.getMap(hazelcastInstance)
+
+    override fun lookupRole(aclKey: AclKey): Role {
+        val principal = principals.getValue(aclKey)
+        return lookupRole(principal.principal)
```

Comment:
This does a bunch of extra work (since `principal` is effectively what you're returning, and the follow up call will do an extra predicate on the mapstore to re-retrieve it) -- would be faster/more straightforward to do something like

```
private fun castSecurablePrincipalAsRole(sp: SecurablePrincipal): Role {
  require(sp.principal.type == PrincipalType.ROLE) { "The provided principal $sp is not a role" }
  return sp as Role
}

override fun lookupRole(principal: Principal): Role {
  val sp = getFirstSecurablePrincipal(findPrincipal(principal))
  return castSecurablePrincipalAsRole(sp)
}

override fun lookupRole(aclKey: AclKey): Role {
  val principal = principals.getValue(aclKey)
  return castSecurablePrincipalAsRole(sp)
}
```

Reply:
Nice catch. I pulled this straight from HazelcastPrincipalService so I'll update these portions